ui/components/Drawers/TransformationsDrawer/index.tsx

```diff
@@ -47,29 +46,37 @@ function TransformationsDrawer({
 		let aiTransformation = [];
 		let basicTransformations = [];
 
-		Object.entries(plugins).map(([, plugin]) => {
-			plugin.operations.map((op) => {
-				if (plugin?.name !== "Basic" && !plugin.credentials?.required)
-					return (aiTransformation = [...aiTransformation, { plugin, op }]);
-				else if (plugin?.name === "Basic") {
-					return (basicTransformations = [
-						...basicTransformations,
-						{ plugin, op },
-					]);
-				}
-			});
+		Object.entries(plugins).forEach(([_, plugin]) => {
+			if (allowedPlugins?.[plugin.identifier] !== undefined) {
+				plugin?.operations?.forEach((op) => {
+					if (
+						allowedPlugins?.[plugin?.identifier]?.operations?.includes(
+							op.displayName
+						)
+					) {
+						if (plugin?.name !== "Basic" && !plugin.credentials?.required)
+							aiTransformation = [...aiTransformation, { plugin, op }];
+						else if (plugin?.name === "Basic") {
+							basicTransformations = [...basicTransformations, { plugin, op }];
+						}
+					}
+				});
+			}
 		});
 
 		const updatedTransformationList =
 			sortTransformationByName(aiTransformation);
 
-		// the first four sequence is fixed for the AI transformation list
-		const identifierSequence = ["wm", "wmc", "erase", "sr"];
+		// the first three sequence is fixed for the AI transformation list
+		const identifierSequence = ["wm", "erase", "sr"];
```

Comment:
bring all constants of file in separate constant JS file and consume it from there.